src/lib.rs

```diff
@@ -326,41 +336,41 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		
 		/// Verify and write a pulse from the beacon into the runtime
 		#[pallet::call_index(0)]
 		#[pallet::weight(T::WeightInfo::write_pulse())]
-		pub fn write_pulse(
-			origin: OriginFor<T>, 
-			pulse: Pulse,
-		) -> DispatchResult {
+		pub fn write_pulse(origin: OriginFor<T>, pulse: Pulse) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-		
-			if let Some(config) = BeaconConfig::<T>::get() {
-				let is_verified = T::Verifier::verify(config, pulse.clone())
-					.map_err(|s| {
+
+			match BeaconConfig::<T>::get() {
+				Some(config) => {
+					let is_verified = T::Verifier::verify(config, pulse.clone()).map_err(|s| {
 						log::error!("Could not verify the pulse due to: {}", s);
 						Error::<T>::PulseVerificationError
 					})?;
-				
-				if is_verified {
-					let current_block = frame_system::Pallet::<T>::block_number();
-					
-					// Retrieve the last pulse index and verify the round number
-					if let Some(last_pulse) = Pulses::<T>::get(current_block - One::one()) {
-						frame_support::ensure!(
-							last_pulse.round < pulse.round, 
-							Error::<T>::InvalidRoundNumber
-						);
+
+					if is_verified {
+						let current_block = frame_system::Pallet::<T>::block_number();
+
+						// Retrieve the last pulse index and verify the round number
+						if let Some(last_pulse) = Pulses::<T>::get(current_block - One::one()) {
+							frame_support::ensure!(
+								last_pulse.round < pulse.round,
+								Error::<T>::InvalidRoundNumber
+							);
+						}
+
+						// Store the new pulse
+						Pulses::<T>::insert(current_block, pulse.clone());
+						// Emit event for new pulse
+						Self::deposit_event(Event::NewPulse { round: pulse.round, who });
 					}
-		
-					// Store the new pulse
-					Pulses::<T>::insert(current_block, pulse.clone());	
-					// Emit event for new pulse
-					Self::deposit_event(Event::NewPulse { round: pulse.round, who });
-				}
+				},
+				None => {
+					log::warn!("No beacon config available");
+				},
 			}
```

Comment:
sorry for the confusing diff, everything but this is just formatting after adding the rustfmt rule set